build/gen_stub.php

```diff
@@ -801,21 +942,67 @@ function(FuncInfo $funcInfo) use(&$generatedFuncInfos) {
     );
 
     if ($fileInfo->generateFunctionEntries) {
-        $code .= "\n\n";
-        $generatedDeclarations = [];
-        $code .= generateCodeWithConditions($funcInfos, "", function(FuncInfo $funcInfo) use (&$generatedDeclarations) {
-            $name = $funcInfo->alias ?? $funcInfo->name;
-            $key = "$name|$funcInfo->cond";
-            if (isset($generatedDeclarations[$key])) {
-                return null;
+        $code .= generateFunctionEntry(null, $fileInfo->funcInfos);
+
+        foreach ($fileInfo->classInfos as $classInfo) {
+            $code .= generateFunctionEntry($classInfo->name, $classInfo->funcInfos);
+        }
+    }
+
+    return $code;
+}
+
+/** @param FuncInfo[] $funcInfos */
+function generateFunctionEntry(?string $className, array $funcInfos): string {
+    $code = "";
+    $generatedDeclarations = [];
+    $code .= generateCodeWithConditions($funcInfos, "", function (FuncInfo $funcInfo) use (&$generatedDeclarations) {
```

Comment:
I think it would make sense to do this part as a single loop over `$fileInfo->getAllFuncInfos()`, i.e. generate all the forward declarations in one block, rather than per-class. Especially because aliases may be referenced from multiple classes in the file, in which case the forward declarations would currently get duplicated.